src/HookTarget/HookTargetGuardian.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {AccessControl} from "openzeppelin-contracts/access/AccessControl.sol";
+import {IHookTarget} from "evk/interfaces/IHookTarget.sol";
+
+/// @title HookTargetGuardian
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice A contract that allows to pause operations that are hooked for the vaults that have this contract installed
+/// as a hook target. The operations remain paused temporarily until either the PAUSE_DURATION elapses or the guardian
+/// calls the unpause function, whichever occurs first.
+contract HookTargetGuardian is IHookTarget, AccessControl {
+    /// @notice Indicates whether the vault operations are currently paused.
+    bool paused;
+
+    /// @notice The timestamp of the last pause.
+    uint48 lastPauseTimestamp;
+
+    /// @notice Role identifier for the guardian role.
+    bytes32 public constant GUARDIAN_ROLE = keccak256("GUARDIAN_ROLE");
+
+    /// @notice The duration for which vaults remain paused.
+    uint256 public immutable PAUSE_DURATION;
+
+    /// @notice The cooldown period before vaults can be paused again.
+    uint256 public immutable PAUSE_COOLDOWN;
+
+    /// @notice Event emitted when the vaults are paused.
+    event Paused();
+
+    /// @notice Event emitted when the vaults are unpaused.
+    event Unpaused();
+
+    /// @notice Error thrown when the vault is paused.
+    error HTG_VaultPaused();
+
+    /// @notice Constructor to initialize the contract with the given admin, pause duration, and pause cooldown.
+    /// @param admin The address of the initial admin.
+    /// @param pauseDuration The duration for which the vault remains paused.
+    /// @param pauseCooldown The cooldown period before the vault can be paused again.
+    constructor(address admin, uint256 pauseDuration, uint256 pauseCooldown) {
+        require(pauseDuration > 0 && pauseCooldown > pauseDuration, "constructor error");
+
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        PAUSE_DURATION = pauseDuration;
+        PAUSE_COOLDOWN = pauseCooldown;
+    }
+
+    /// @notice Fallback function to revert if the vault is paused.
+    fallback() external {
+        if (isPaused()) revert HTG_VaultPaused();
```

Comment:
You could probably do selective re-enabling of operations with a mapping that contains "whitelisted" `bytes4` selectors. If paused, check if the selector exists in the mapping. Only if not do you revert.

Reply:
Well, I believe you mentioned we're not planning on using the hook target version so I wouldn't bother. If you already have the code maybe throw it in for the audit, but I wouldn't do anything special.